src/go/rpk/pkg/cli/cluster/partitions/list.go

```diff
@@ -0,0 +1,221 @@
+package partitions
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-units"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/types"
+)
+
+func newListPartitionMovementsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		completion int
+		all        bool
+		human      bool
+		partitions []string
+		response   []adminapi.ReconfigurationsResponse
+	)
+	cmd := &cobra.Command{
+		Use:     "move-status",
+		Aliases: []string{"alter-status"},
+		Short:   "Show ongoing partition movements",
+		Long:    helpListMovement,
+		Run: func(cmd *cobra.Command, args []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			// If partition(s) is specified but no topic(s) is specified, exit.
+			if len(args) <= 0 && len(partitions) > 0 {
+				out.Die("specify at least one topic when --partition is used, exiting.")
+			}
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			response, err = cl.Reconfigurations(cmd.Context())
+			out.MaybeDie(err, "unable to list partition movements: %v\n", err)
+
+			if len(response) == 0 {
+				out.Exit("Currently there's no ongoing partition movements.")
+			}
+
+			if len(args) > 0 && len(partitions) > 0 { // yes topic && yes partition
+				j := 0
+				for _, t := range args {
+					if strings.Count(t, "/") == 0 {
+						for _, p := range partitions {
+							for _, res := range response {
+								if res.Ns == "kafka" && res.Topic == t && strconv.Itoa(res.PartitionID) == p {
+									response[j] = res
+									j++
+								}
+							}
+						}
+					} else if strings.Count(t, "/") == 1 {
+						nt := strings.Split(t, "/")
+						for _, p := range partitions {
+							for _, res := range response {
+								if res.Ns == nt[0] && res.Topic == nt[1] && strconv.Itoa(res.PartitionID) == p {
+									response[j] = res
+									j++
+								}
+							}
+						}
+					} else {
+						fmt.Printf("invalid format for topic %s, skipping.\n", t)
+					}
+				}
+				response = response[:j]
+			} else if len(args) > 0 && len(partitions) <= 0 { // yes topic && no partition
+				j := 0
+				for _, t := range args {
+					if strings.Count(t, "/") == 0 {
+						for _, res := range response {
+							if res.Ns == "kafka" && res.Topic == t {
+								response[j] = res
+								j++
+							}
+						}
+					} else if strings.Count(t, "/") == 1 {
+						nt := strings.Split(t, "/")
+						for _, res := range response {
+							if res.Ns == nt[0] && res.Topic == nt[1] {
+								response[j] = res
+								j++
+							}
+						}
+					} else {
+						fmt.Printf("invalid format for topic %s, skipping.\n", t)
+					}
+				}
+				response = response[:j]
+			}
```

Comment:
wdyt of splitting the args:
```
// Split the topic into namespace and topic name if it contains a '/'
topicParts := strings.Split(t, "/")

if len(topicParts) > 1 {

}
```

Also, you don't need to check for the length of `args` if it's empty the `for loop` will be a no-op. I also think you can loop the response and check if the partition slice contains the given res.PartitionID. Maybe using a helper function like:


```
func contains(slice []string, s string) bool {
	for _, str := range slice {
		if str == s {
			return true
		}
	}
	return false
}
```

So I imagine we could do something like:

```
var response []adminapi.ReconfigurationsResponse
for _, t := range args {
  for _, r := range response {
     nt := strings.Split(t, "/")

     isKafkaTopic := len(topicParts) == 1 && res.Ns == "kafka" && res.Topic == t 
     isNamespaced := len(topicParts) == 2 && res.Ns == nt[0] && res.Topic == nt[1] 
     
     // Then here you do the partition check using the helper
     if isKafka || isNamespaced {
       if len(partitions) <= 0 || contains(partitions, strconv.Itoa(res.PartitionID)) {
         response = append(response, r)
       }
     }
  }
}
```

what do you think?

Reply:
Cool. It is much simpler :)